### PR TITLE
Add type param to `xml:get` return type

### LINKED ISSUE #33407: `xml:get(xml<ItemType> x, int i)` return type is different from spec
**Description:**
the spec says it should return `ItemType` but the implementation returns `xml`

**Steps to reproduce:**

Spec:
```
# Returns the item of `x` with index `i`.
# This differs from `x[i]` in that it panics if
# `x` does not have an item with index `i`.
#
# + x - the xml sequence 
# + k - the index
# + return - the item with index `i` in `x`
public isolated function get(xml<ItemType> x, int i) returns ItemType = external;
```
Implementation
```
# Returns the item of an xml sequence with given index.
#
# This differs from `x[i]` in that it panics if
# parameter `x` does not have an item with index parameter `i`.
#
# + x - the xml sequence
# + i - the index
# + return - the item with index parameter `i` in parameter `x`
public isolated function get(xml<ItemType> x, int i) returns xml = @java:Method {
    'class: "org.ballerinalang.langlib.xml.Get",
    name: "get"
} external;
```

**Affected Versions:**

**OS, DB, other environment details and versions:**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

### PR DESCRIPTION
**Pull request by KRVPerera**
## Purpose

Fixes #33407

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Comment by MaryamZi** [comment]
Workflows have been enabled on the langlib-fixes branch and some existing failures have been fixed. Shall we sync this PR also anyway?

File: tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/types/xml/XMLIterationTest.java [hunk 1]
```diff
@@ -61,59 +61,48 @@ public void testNegative() {
         BAssertUtil.validateError(negative, index++,
                 "incompatible types: 'xml:Element' is not an iterable collection",
                 29, 34);
-        BAssertUtil.validateError(negative, index++,
-                "incompatible types: expected 'record {| xml:Element value; |}?', found 'record " +
```
[the remark this answers is not in the document]
Comment (KRVPerera) [reply tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/types/xml/XMLIterationTest.java:65]: Fixed 590900b41c158f70f9a3ee13531ed70c9efc6021

**Review by nadeeshaan — APPROVED** [review]
Approving on behalf of the LS changes

File: tests/ballerina-compiler-api-test/src/test/java/io/ballerina/semantic/api/test/langlib/TypeParamBoundXMLFunctionsTest.java [hunk 2]
```diff
@@ -85,7 +85,7 @@ public void testGet() {
         assertTrue(getFnType.returnTypeDescriptor().isPresent());
 
         TypeSymbol getFnRetType = getFnType.returnTypeDescriptor().get();
-        assertEquals(getFnRetType.typeKind(), TypeDescKind.XML);
```
[the remark this answers is not in the document]
Comment (pcnfernando) [reply tests/ballerina-compiler-api-test/src/test/java/io/ballerina/semantic/api/test/langlib/TypeParamBoundXMLFunctionsTest.java:88]: This change is expected AFIK.
@mohanvive @malinthar can you please confirm

[the remark this answers is not in the document]
Comment (malinthar) [reply tests/ballerina-compiler-api-test/src/test/java/io/ballerina/semantic/api/test/langlib/TypeParamBoundXMLFunctionsTest.java:88]: @pcnfernando Yes. This change is expected!

**Pull request merged**